GuidedTutorials/HeatEquation/Source/main.py

```diff
@@ -0,0 +1,141 @@
+import numpy as np
+import amrex.space3d as amr
+
+def main_main ():
+    """
+    TODO Add documentation
+    """
+    # AMREX_D_DECL means "do the first X of these, where X is the dimensionality of the simulation"
+    dom_lo = amr.IntVect(*amr.d_decl(       0,        0,        0))
+    dom_hi = amr.IntVect(*amr.d_decl(n_cell-1, n_cell-1, n_cell-1))
+
+    # Make a single box that is the entire domain
+    domain = amr.Box(dom_lo, dom_hi)
+
+    # Make BoxArray and Geometry:
+    # ba contains a list of boxes that cover the domain,
+    # geom contains information such as the physical domain size,
+    # number of points in the domain, and periodicity
+
+    # Initialize the boxarray "ba" from the single box "domain"
+    ba = amr.BoxArray(domain)
+    # Break up boxarray "ba" into chunks no larger than "max_grid_size" along a direction
+    ba.max_size(max_grid_size)
+
+    # This defines the physical box, [0,1] in each direction.
+    real_box = amr.RealBox([*amr.d_decl( 0., 0., 0.)], [*amr.d_decl( 1., 1., 1.)])
+
+    # This defines a Geometry object
+    # periodic in all direction
+    coord = 0 # Cartesian
+    is_per = [*amr.d_decl(1,1,1)] # periodicity
+    geom = amr.Geometry(domain, real_box, coord, is_per);
+
+    # Extract dx from the geometry object
+    dx = geom.data().CellSize()
+
+    # Nghost = number of ghost cells for each array
+    Nghost = 1
+
+    # Ncomp = number of components for each array
+    Ncomp = 1
+
+    # How Boxes are distrubuted among MPI processes
+    dm = amr.DistributionMapping(ba)
+
+    # Allocate two phi multifabs: one will store the old state, the other the new.
+    phi_old = amr.MultiFab(ba, dm, Ncomp, Nghost)
+    phi_new = amr.MultiFab(ba, dm, Ncomp, Nghost)
+    phi_old.set_val(0.)
+    phi_new.set_val(0.)
+
+    # time = starting time in the simulation
+    time = 0.
+
+    # Ghost cells
+    ng = phi_old.nGrowVect
+    ngx = ng[0]
+    ngy = ng[1]
+    ngz = ng[2]
+
+    # Loop over boxes
+    for mfi in phi_old:
+        bx = mfi.validbox()
+        # phiOld is indexed in reversed order (z,y,x) and indices are local
+        phiOld = np.array(phi_old.array(mfi), copy=False)
```

Comment:
@RevathiJambunathan are you still pushing the `cupy`/`numpy` generalization we worked on last Friday? :)

Reply:
I pushed this now :)